File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    var answer = 0
    
    func dfs(_ start : Int, _ index : Int) {
        if index == numbers.count {
            if start == target {
                answer += 1
            }
            return
        }
        dfs(start + numbers[index],index+1)
        dfs(start - numbers[index],index+1)
    }
    dfs(0,0)
    
    
    return answer
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0
def solution(numbers, target):
    n = len(numbers)
    def dfs(idx,cur):
        global answer
        if idx == n :
            if cur == target:
                answer += 1
            return
        dfs(idx+1,cur+numbers[idx])
        dfs(idx+1,cur-numbers[idx])
    dfs(0,0)
    
    return answer

            